### PR TITLE
G 68 notification from HIU  to CM via gateway

### PR DESCRIPTION
**Pull request by gayatrijagtap**

### CODE REVIEW

**Review by dheeraj-p — APPROVED** [review]

File: src/main/java/in/org/projecteka/hiu/dataprocessor/HealthDataProcessor.java [hunk 1]
```diff
@@ -165,23 +166,40 @@ private void notifyHealthInfoStatus(DataContext context,
                                         SessionStatus sessionStatus) {
         String consentId = dataFlowRepository.getConsentId(context.getTransactionId()).block();
         String hipId = consentRepository.getHipId(consentId).block();
-        HealthInfoNotificationRequest healthInfoNotificationRequest = HealthInfoNotificationRequest.builder()
+        HealthInfoNotificationRequest healthInfoNotificationRequest =
+                getHealthInfoNotificationRequest(context, statusResponses, sessionStatus, consentId, hipId);
+        String token = centralRegistry.token().block();
```
Comment (dheeraj-p) [inline src/main/java/in/org/projecteka/hiu/dataprocessor/HealthDataProcessor.java:171]: Is blocking fine here?

File: src/main/java/in/org/projecteka/hiu/dataprocessor/HealthDataProcessor.java [hunk 2]
```diff
@@ -165,23 +166,40 @@ private void notifyHealthInfoStatus(DataContext context,
...
                         .hipId(hipId)
                         .statusResponses(statusResponses)
                         .build())
                 .build();
-        String token = centralRegistry.token().block();
-        healthInformationClient.notifyHealthInfo(healthInfoNotificationRequest, token).block();
+    }
+
+    private String fetchCMId(String consentId) {
+        return consentRepository.getConsentMangerId(consentId).block();
```
Comment (dheeraj-p) [inline src/main/java/in/org/projecteka/hiu/dataprocessor/HealthDataProcessor.java:202]: Same as above

File: src/main/java/in/org/projecteka/hiu/dataprocessor/HealthDataProcessor.java (hunk 1 above)
Comment (gayatrijagtap) [reply src/main/java/in/org/projecteka/hiu/dataprocessor/HealthDataProcessor.java:171]: the method notifyHealthInfoStatus is not in reactive context. So, we didn't find any other way than blocking

File: src/main/java/in/org/projecteka/hiu/dataprocessor/HealthDataProcessor.java (hunk 2 above)
Comment (gayatrijagtap) [reply src/main/java/in/org/projecteka/hiu/dataprocessor/HealthDataProcessor.java:202]: same reply

**Pull request merged**